presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -849,4 +862,65 @@ private static String typedNull(String prestoType)
     {
         return format("CAST(NULL AS %s)", prestoType);
     }
+
+    private static List<ColumnInfo> getDescribeOutputColumnInfoList(ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = getClientTypeSignatureFromTypeString(resultSet.getString("Type"));
+            ColumnInfo.Builder builder = new ColumnInfo.Builder()
+                    .setColumnName(columnName)
+                    .setColumnLabel(columnName)
+                    .setCatalogName(catalog)
+                    .setSchemaName(schema)
+                    .setTableName(table)
+                    .setColumnTypeSignature(clientTypeSignature)
+                    .setNullable(ColumnInfo.Nullable.UNKNOWN);
+            setTypeInfo(builder, clientTypeSignature);
+            list.add(builder.build());
+        }
+        return list.build();
+    }
+
+    @VisibleForTesting
+    static ClientTypeSignature getClientTypeSignatureFromTypeString(String type)
+    {
+        Matcher topLevelMatcher = topLevelTypeRegexPattern.matcher(type);
```

Comment:
Move this down so that it's directly before the if statement, since it's specific to just that if block. The other variables are for the entire method.